### PR TITLE
[#25] Remove sys-info routes

### LINKED ISSUE #25: Remove sys-info routes
As stated in https://github.com/meilisearch/MeiliSearch/pull/810 and https://github.com/meilisearch/integration-guides/issues/36 the sys-info routes will be considered as deprecated in MeiliSearch v0.13.0. These routes should be removed from this SDK accordingly.

### PR DESCRIPTION
**Pull request by Ethenyl**
# Summary

Remove deprecated sys-info routes.

Also performed minor code cleanup regarding the use of the `JSONDecoder`.

# Miscellaneous

Fixes #25.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,9 @@
 #
 # gitignore contributors: remember to update Global/Xcode.gitignore, Objective-C.gitignore & Swift.gitignore
 
+## macOS
+.DS_Store
+
```
Comment (curquiza) [inline .gitignore:7]: Why do you remove this in the gitignore? 🙂

Comment (curquiza) [inline .gitignore:7]: Why do you think it's better remove this? 🙂

**Review by curquiza — COMMENTED** [review]
LGTM! The code cleanup seems nice 🙂

File: .gitignore (hunk 1 above)
Comment (curquiza) [reply .gitignore:7]: In fact, it prevents contributors to accidentally add the `.DS_Store` file, and to waste time asking them to remove it before merging. When everything else is ok, it's a little bit frustrating for everyone 😉

**Review by curquiza — APPROVED** [review]

File: Sources/MeiliSearch/System.swift [hunk 2]
```diff
@@ -8,138 +8,70 @@ struct System {
...
 
-    func health(_ completion: @escaping (Result<(), Swift.Error>) -> Void) {
-
-        self.request.get(api: "/health") { result in
+    // MARK: Functions
 
+    func health(_ completion: @escaping (Result<(), Swift.Error>) -> Void) {
+        request.get(api: "/health") { result in
             switch result {
+
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/System.swift:20]:
```suggestion
```

**Review by bidoubiwa — COMMENTED** [review]
Thanks a lot!

File: Sources/MeiliSearch/System.swift [hunk 3]
```diff
@@ -8,138 +8,70 @@ struct System {
...
-                    let vesion: Version = try decoder.decode(Version.self, from: data)
-                    completion(.success(vesion))
-                } catch {
-                    completion(.failure(error))
-                }
-
-            case .failure(let error):
-                completion(.failure(error))
-            }
+    func updateHealth(_ health: Bool, _ completion: @escaping (Result<(), Swift.Error>) -> Void) {
```
[the remark this answers is not in the document]
Comment (eskombro) [reply Sources/MeiliSearch/System.swift:29]: We can either remove it here, or create an issue for a further PR

**Review by eskombro — CHANGES_REQUESTED** [review]

File: Sources/MeiliSearch/System.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ethenyl) [reply Sources/MeiliSearch/System.swift:29]: I understand your point but I rather not change something unrelated to the issue at hand so a new issue is fine for me.

File: Sources/MeiliSearch/System.swift (hunk 2 above)
Comment (Ethenyl) [reply Sources/MeiliSearch/System.swift]: Fixed in `104dc58`.

File: Sources/MeiliSearch/System.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (curquiza) [reply Sources/MeiliSearch/System.swift:29]: @eskombro The route is still available, and I would rather this to be done in another PR anyway (once the route will be really removed) 🙂

[the remark this answers is not in the document]
Comment (bidoubiwa) [reply Sources/MeiliSearch/System.swift:29]: It will still exist in v0.15, I checked with @curquiza

[the remark this answers is not in the document]
Comment (eskombro) [reply Sources/MeiliSearch/System.swift:29]: Sorry, this was my bad, we had a discussion about it but it hasn't been removed at all from the Core, please ignore my comment, sorry!

[the remark this answers is not in the document]
Comment (eskombro) [reply Sources/MeiliSearch/System.swift:29]: https://github.com/meilisearch/MeiliSearch/issues/940

**Review by eskombro — APPROVED** [review]

**Comment by curquiza** [comment]
Thanks for your review guys, and thanks for your work @Ethenyl! Merging!!

**Comment by Ethenyl** [comment]
@curquiza my pleasure, if help is needed on other topics, I will gladly help during **Hacktober** 👻

**Comment by Ethenyl** [comment]
@bidoubiwa & @eskombro, thanks for your reviews 😉

**Pull request merged**